### PR TITLE
AutomationPractice test suite

### PR DESCRIPTION
**Pull request by CasaccioFacundo**
Created test suite automation-practice.xml and configured it to run from maven all 6 tests at the same time

### CODE REVIEW

**Comment by ankochergin** [comment]
👍

File: src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package com.qaprosoft.carina.demo.utils;
+
+import com.qaprosoft.carina.demo.gui.pages.AutomationPracticeCreateAccountPage;
+import com.qaprosoft.carina.demo.gui.pages.AutomationPracticeSignInPage;
+
+import java.util.Random;
+
+public class AuthService {
+
+    public void signIn(AutomationPracticeSignInPage signInPage, String email, String password) {
```
Comment (ankochergin) [inline src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java]: do not use pages as an argument for method. Define needed pages inside methods. In carina util classes you can find how to get driver here

File: src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        signInPage.typeEmail(email);
+        signInPage.typePassword(password);
+        signInPage.clickSignInButton();
+    }
+
+    public void signIn(AutomationPracticeSignInPage signInPage) {
```
Comment (ankochergin) [inline src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java]: I like this method with randomly generated email and password. Imagine that you need to use this email and password somewhere else in the test. I suggest to return email and password to the test

File: src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import static com.zebrunner.agent.core.registrar.TestRunRegistrar.LOGGER;
+import static com.zebrunner.agent.core.webdriver.RemoteWebDriverFactory.getDriver;
+
+public class ScreenshotService {
```
Comment (ankochergin) [inline src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java]: code doesn't compile. never push not working code

File: src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    File outputfile = new File("/somePath/test1.png");
+    BufferedImage screenshot = null;
```
Comment (ankochergin) [inline src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java]: define this inside methods

File: src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java (hunk 3 above)
Comment (ankochergin) [reply src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java]: I suggest to store path in testdata

File: src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import com.qaprosoft.carina.demo.gui.pages.AutomationPracticeCreateAccountPage;
+import com.qaprosoft.carina.demo.gui.pages.AutomationPracticeSignInPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+
+import java.util.Random;
+
+public class AuthService implements IDriverPool {
+
+    private WebDriver getDriverSafe() {
```
Comment (ankochergin) [inline src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java]: why you decided to implement here getDriverSafe()?

File: src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Date;
+
+import static com.zebrunner.agent.core.registrar.TestRunRegistrar.LOGGER;
+
+public class ScreenshotService implements IDriverPool {
+
+    private final File directory = new File("src/test/resources/testdata");
```
Comment (ankochergin) [inline src/main/java/com/qaprosoft/carina/demo/utils/ScreenshotService.java]: lets store path in testdata

File: src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java (hunk 5 above)
Comment (CasaccioFacundo) [reply src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java]: I tried to use getDriver() and didnt work for me, so I looked up in a util class that is in the carina demo and saw that method, tried it and work. But it seem that I was doing something wrong, now I refactored to just getDriver() and works perfectly.

**Pull request merged**